elasticsearch/src/main/java/org/hibernate/ogm/datastore/elasticsearch/ElasticSearchDialect.java

```diff
@@ -0,0 +1,195 @@
+import org.hibernate.ogm.model.spi.Tuple;
+
+public class ElasticSearchDialect extends BaseGridDialect {
+
+	private final ElasticSearchDatastoreProvider provider;
+
+	private final Set<String> sequences = new HashSet<>();
+
+	public ElasticSearchDialect(ElasticSearchDatastoreProvider provider) {
+		super();
+		this.provider = provider;
+	}
+
+	@Override
+	public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
+		GetResponse getResponse = provider.getClient().prepareGet(provider.getDatabase(), key.getTable(), id(key)).setOperationThreaded(false).execute().actionGet();
+		if (getResponse != null) {
+			return new Tuple(new MapTupleSnapshot(getResponse.getSource()));
+		}
+		return null;
+	}
+
+	private String id(EntityKey key) {
+		return String.valueOf(key.getColumnValues()[0]);
+	}
+
+	@Override
+	public Tuple createTuple(EntityKey key, TupleContext tupleContext) {
+		Map<String, Object> map = map(key);
+		return new Tuple(new MapTupleSnapshot(map));
+	}
+
+	private Map<String, Object> map(EntityKey key) {
+		Map<String, Object> values = new HashMap<>();
+		for (int i = 0; i < key.getColumnNames().length; i++) {
+			values.put(key.getColumnNames()[i], key.getColumnValues()[i]);
+		}
+		return values;
+	}
+
+	@Override
+	public void insertOrUpdateTuple(EntityKey key, Tuple tuple, TupleContext tupleContext) throws TupleAlreadyExistsException {
+		Map<String, Object> map = Maps.newHashMap(((MapTupleSnapshot) tuple.getSnapshot()).getMap());
+		MapHelpers.applyTupleOpsOnMap(tuple, map);
+		IndexRequest upsert = new IndexRequest(provider.getDatabase(), key.getTable(), id(key)).source(map).listenerThreaded(false);
+		provider.getClient().prepareUpdate(provider.getDatabase(), key.getTable(), id(key)).setDoc(map).setUpsert(upsert).execute().actionGet();
+	}
+
+	@Override
+	public void removeTuple(EntityKey key, TupleContext tupleContext) {
+		provider.getClient().prepareDelete(provider.getDatabase(), key.getTable(), id(key)).setOperationThreaded(false).setRefresh(true).execute().actionGet();
+	}
+
+	@Override
+	public Association getAssociation(AssociationKey key, AssociationContext associationContext) {
+		return null;
+	}
+
+	@Override
+	public Association createAssociation(AssociationKey key, AssociationContext associationContext) {
+		return null;
+	}
+
+	@Override
+	public void insertOrUpdateAssociation(AssociationKey key, Association association, AssociationContext associationContext) {
+
+	}
+
+	@Override
+	public void removeAssociation(AssociationKey key, AssociationContext associationContext) {
+
+	}
+
+	@Override
+	public boolean isStoredInEntityStructure(AssociationKeyMetadata associationKeyMetadata, AssociationTypeContext associationTypeContext) {
+		return false;
+	}
+
+	@Override
+	public Number nextValue(NextValueRequest request) {
+		String sequencesIndex = databaseSequencesIndex();
+		if (!sequences.contains(request.getKey().getTable())) {
+			final Map<String, Object> disabled = new HashMap<String, Object>() {
```

Comment:
I would prefer to not use this pattern, as it's creating anonymous classes. In this case, also just `Collections#singletonMap()` can be used.